chips/apollo3/src/iom.rs

```diff
@@ -499,14 +500,25 @@ impl<'a> Iom<'_> {
 
                 self.read_data();
             }
-        }
 
-        if irqs.is_set(INT::CMDCMP) {
+            // The IOM doesn't work very well when using non-blocking operations
+            // without the DMA.
+            // We don't currently support the DMA, so we hit some weird interrupt
+            // corner cases. So make the IOM more stable, let's wait for the idle
+            // status here.
+            while regs.status.read(STATUS::IDLESET) != 1 {}
```

Comment:
How long does this take? If it's more than 100us this should be non-blocking. What does "doesn't work very well" mean?

Reply:
It should be instantaneous. We are in the interrupt for the command completed, so we should be in the idle state.

I have just seen cases where it can take a few us for the state to change back to idle.

"doesn't work very well" means that we can drop operations as we write the data before the hardware is idle